### PR TITLE
Remove timeslice support

### PR DESCRIPTION
**Pull request by lucasdemarchi**
This has been deprecated by the thread per bus infra.

### CODE REVIEW

**Comment by tridge** [comment]
makes sense, looks good to me, just need a pass in travis

**Comment by jberaud** [comment]
Makes sense. Ok for me.

File: libraries/AP_Compass/AP_Compass_AK8963.cpp [hunk 1]
```diff
@@ -272,7 +272,7 @@ void AP_Compass_AK8963::_update_timer()
 {
     uint32_t now = AP_HAL::micros();
 
-    if (!_timesliced && now - _last_update_timestamp < 10000) {
```
Comment (OXINARF) [inline libraries/AP_Compass/AP_Compass_AK8963.cpp]: Variable hasn't been removed from class definition.

File: libraries/AP_Compass/AP_Compass_AK8963.cpp [hunk 2]
```diff
@@ -170,7 +170,7 @@ bool AP_Compass_AK8963::init()
     /* timer needs to be called every 10ms so set the freq_div to 10 */
     if (!_bus->register_periodic_callback(10000, FUNCTOR_BIND_MEMBER(&AP_Compass_AK8963::_update, void))) {
         // fallback to timer
-        _timesliced = hal.scheduler->register_timer_process(FUNCTOR_BIND_MEMBER(&AP_Compass_AK8963::_update_timer, void), 10);
+        hal.scheduler->register_timer_process(FUNCTOR_BIND_MEMBER(&AP_Compass_AK8963::_update_timer, void));
```
Comment (OXINARF) [inline libraries/AP_Compass/AP_Compass_AK8963.cpp]: Do we still want to do this fallback? I think this is the only place I'm aware where we care about the return value of *register_periodic_callback*.

File: libraries/AP_Compass/AP_Compass_AK8963.cpp (hunk 1 above)
Comment (lucasdemarchi) [inline libraries/AP_Compass/AP_Compass_AK8963.cpp]: Thanks, will do

File: libraries/AP_Compass/AP_Compass_AK8963.cpp (hunk 2 above)
Comment (lucasdemarchi) [inline libraries/AP_Compass/AP_Compass_AK8963.cpp]: I had the same feeling. I don't quiet remember why the fallback has been added.... I think it was for PX4 before it had the bus per thread, but I may be wrong.

**Comment by lucasdemarchi** [comment]
Force pushed with the removal of unused var suggested by @OXINARF. Thanks.

Regarding the fallback to timer thread, I'll leave it for another PR since I'm not sure about its implications.

**Comment by OXINARF** [comment]
> Regarding the fallback to timer thread, I'll leave it for another PR since I'm not sure about its implications.

If you can open a PR doing that, it would be great, otherwise it will just be forgotten.

Besides that, this looks good to me.

**Comment by lucasdemarchi** [comment]
I was doing it... now I see why the fallback is there. The register_periodic_callback() on AuxiliaryBus is not implemented.

**Comment by lucasdemarchi** [comment]
This also means that mpu6000 + hmc5883 was broken. I'll open a PR.

**Pull request merged**